src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/AssemblyBasedCallerUtils.java

```diff
@@ -187,6 +188,10 @@ public static ReadThreadingAssembler createReadThreadingAssembler(final Assembly
         return assemblyEngine;
     }
 
+    public static HaplotypeBAMWriter createBamWriter(final AssemblyBasedCallerArgumentCollection args, final SAMFileHeader header) {
```

Comment:
Instead of returning `null` when bamout path is not defined, we should use an `Optional`. So we will have something like:

```java

public static Optional<HaplotypeBAMWriter> createBamWriter(final AssemblyBasedCallerArgumentCollection args, final SAMFileHeader header) {
        return args.bamOutputPath != null ? Optional.of(HaplotypeBAMWriter.create(args.bamWriterType, new File(args.bamOutputPath), header)) : Optional.empty();
}

```

This clearly signals to the reader that the variable is empty sometimes.